packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -142,6 +142,59 @@ class _AnimatedState extends State<AnimatedWidget> {
   Widget build(BuildContext context) => widget.build(context);
 }
 
+/// Signature for a convenience function for passing around a builder for a
+/// transiton's secondary animation.
+///
+/// Typically used as the argument for [DelegatedTransition.builder].
+typedef DelegatedTransitionBuilder = Widget Function(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget? child);
+
+/// An exit animation for a route that can be provided to another route.
+///
+/// {@template flutter.widgets.delegatedTransition}
+/// Used for the purposes of coordinating transitions between two routes with
+/// different route transitions. When a route is added to the stack, the original
+/// topmost route will look for this transition, and if diferent from its own
+/// `DelegatedTransition`, it will use this [builder] to animate off the screen.
+/// {@endtemplate}
+///
+/// This check is based on the run time type of the `DelegatedTransition`. So if
+/// the topmost route has a `DelegatedTransition` of the same type as the route
+/// below it, then the transition will not be used in the lower route, even if
+/// the builder is different.
+///
```

Comment:
Nit: Maybe somewhere explain that typically you would subclass this, override `builder`, and pass an instance to PageRoute.delegatedTransition?